atoma-proxy/src/server/middleware.rs

```diff
@@ -749,6 +748,30 @@ pub(crate) mod auth {
                     return Err(StatusCode::NOT_FOUND);
                 }
             };
+            // This will fail if the balance is not enough.
```

Comment:
This should never happen right ?

Reply:
My concern is that we are buying an exceeding limit of tokens, and the user/client only has paid for a lower number of tokens

Reply:
The postgres check doesn't allow you to go into negative values. So you never buy more than what the user actually have in the account.

Reply:
My point is, when we buy a new stack. We are using the total proxy wallet amount, which is the sum of all users payments. Imagine now we are spending more for each user than the user has paid (since we always buy max_compute_units for each node, which is rn hardcoded to 2.56M compute units). I am afraid this could lead to burn funds without users having paid for it

Reply:
the amount of usdc that proxy spends when acquiring new stack is first deducted from the users balance. If the deduction fails nothing is acquired.

Reply:
I see, but then this is a problem too. Imagine you charge $1 per 1M tokens, then if a user submits $10, we can at most use 3x$2.56, and the rest we need to either send back to the user, but we can't use it to buy more compute.

Reply:
if user sends $10 and use 3x$2.56 he will have $2.52 in the account left.  I don't see the problem. For now we don't support withdrawal. But we can add that in the future.